### DISCUSSION TITLE
This kind of robust optimization model cannot be solved

### DISCUSSION THREAD

**Discussion by WuSiren** [yalmip/YALMIP#1138] [Optimization modelling]
This kind of robust optimization model cannot be solved:
```
c = [1; 5];
x = sdpvar(2,1);
w = sdpvar(2,1);
y = sdpvar(1);
F = [x' * w == y, 0 <= x <= 10, 5 <= y <= 10];
W = [[0.4; 0.5] <= w <= [0.6; 0.7], uncertain(w)];
objective = c' * x;
sol = optimize(F + W,objective);
```
But if I remove the equality constraint, it can be solved normally: 
```
c = [1; 5];
x = sdpvar(2,1);
w = sdpvar(2,1);
F = [0 <= x <= 10, 5 <= x' * w <= 10];
W = [[0.4; 0.5] <= w <= [0.6; 0.7], uncertain(w)];
objective = c' * x;
sol = optimize(F + W,objective);
```
However, in practical applications, it is inevitable that equality constraints will appear, and in many cases it is not easy to simplify. How to perform robust optimization?

**Comment by johanlofberg** [comment]
https://yalmip.github.io/equalityinuncertainty

**Comment by WuSiren** [comment]
> https://yalmip.github.io/equalityinuncertainty

Thank you! This is the clearest discussion on this issue I have ever seen.

**Comment by WuSiren** [comment]
> https://yalmip.github.io/equalityinuncertainty

I have a **deterministic (not robust) mixed integer linear programming** model where there is an equality constraint. I can treat the equality constraint simply as an assignment from fixed decisions, such as
```
...
QI = sdpvar(10);
Q = sdpvar(3,10);
for n = 1:10
    QI(n) = sum(Q(:,n));
end
constranits = [..., ..., ...];
objective = ...;
options = sdpsettings('verbose',1,'solver','mosek');
sol = optimize(constranits, objective, options);
```
or exactly as an equality constraint, such as
```
...
QI = sdpvar(10);
Q = sdpvar(3,10);
constranits = [..., ..., ...];
for n = 1:10
    constranits = constranits + [QI(n) == sum(Q(:,n))];
end
objective = ...;
options = sdpsettings('verbose',1,'solver','mosek');
sol = optimize(constranits, objective, options);
```
Both expressions can be solved by solver Mosek, but they give different results (the objective value of the former is smaller). I don't see any obvious contradiction in both results, but why are they different? Which is the correct way to write the model?
I'm waiting for your kind help. Thanks!!!

**Comment by johanlofberg** [comment]
Don't use the issue tracker as a discussion board. Use the google groups forum for that
https://groups.google.com/g/yalmip

**Comment by WuSiren** [comment]
> ‎不要将问题跟踪器用作讨论板。使用谷歌组论坛‎
> https://groups.google.com/g/yalmip

Sorry. I will post it again using the google groups forum.

**Comment by WuSiren** [comment]
> Don't use the issue tracker as a discussion board. Use the google groups forum for that
> https://groups.google.com/g/yalmip

I beg your help. 🌹
